src/libsyntax/parse/parser/expr.rs

```diff
@@ -1093,6 +1093,26 @@ impl<'a> Parser<'a> {
         let mut attrs = outer_attrs;
         attrs.extend(self.parse_inner_attributes()?);
 
+        if let Some(ref interp_user_fn) = self.interp_user_fn {
```

Comment:
This should be in it's own method with an associated doc comment describing more clearly what it does.

Also, why do this in the parser instead of a post-processing step on the AST?

Reply:
I've gone back and forth so many times between those two approaches; this basically turned out to be the simplest and most efficient way to do it.

Reply:
I'd personally prefer not to add more mutable state to the parser if possible.

Reply:
This is the opposite. I’d have to make the entire AST mutable if I didn’t do something like this.

Reply:
Oh I see, we mustn't do that.

Reply:
@Centril Yeah, also, I'd need something like a mutable version of `ast::Vistitor` (not the current `MutVisitor`, which operates quite differently) to do that.

Reply:
@alexreg yes yes, just the comment would be good :)

Reply:
@Centril Yeah, also, I'd need something like a mutable version of `ast::Vistitor` (not the current `MutVisitor`, which operates quite differently) to do that.